MCLauncher/MainWindow.xaml.cs

```diff
@@ -228,8 +245,21 @@ public partial class MainWindow : Window, ICommonVersionCommands {
             });
         }
 
-        private void InvokeRemove(Version v) {
-            Directory.Delete(v.GameDirectory, true);
+        private async void InvokeRemove(Version v) {
```

Comment:
By making this async you are introducing a significant issue - the button can now be clicked twice and the function will be invoked twice. TL;Dr you probably don't want that unless you can either add special handling for that case and/or can display the progress to user somehow.

As I believe that displaying progress for removal is out of scope for this PR this should be rewritten to remove the `async` prefix.